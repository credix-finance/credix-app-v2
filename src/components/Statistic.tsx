import React, { useMemo } from "react";

interface StatisticProps {
	label: string;
	value: number;
	currency?: string;
	isPercentage?: boolean;
}

export const Statistic = ({ label, value, currency, isPercentage = false }: StatisticProps) => {
	const formatter = useMemo(
		() =>
			Intl.NumberFormat("en", {
				style: isPercentage ? "percent" : undefined,
				notation: !isPercentage ? "compact" : undefined,
				minimumFractionDigits: 0,
				maximumFractionDigits: 1,
			}),
		[isPercentage]
	);

	const formattedValue = useMemo(() => {
		return formatter.format(value);
	}, [formatter, value]);

	return (
		<div className="bg-credix-primary border border-solid border-darker rounded-[1px] font-sans h-36 w-min min-w-[16rem] md:w-full md:min-w-[12rem] ml-[21.5px] pr-5 flex items-center">
			<div className="bg-credix-primary ml-[-21.5px]">
				<div className="capitalize">{label}</div>
				<div className="text-6xl font-bold">
					{formattedValue}
					<span className="text-sm font-normal">{currency}</span>
				</div>
			</div>
		</div>
	);
};
