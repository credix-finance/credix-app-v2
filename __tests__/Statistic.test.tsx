import React from "react";
import renderer from "react-test-renderer";
import { Statistic } from "@components/Statistic";

test("Currency statistic", () => {
	const props = {
		label: "TVL",
		value: 14800000,
		currency: "USDC",
		isPercentage: false,
	};

	const component = renderer.create(<Statistic {...props} />);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test("Percentage statistic", () => {
	const props = {
		label: "Estimated APY",
		value: 0.138,
		isPercentage: true,
	};

	const component = renderer.create(<Statistic {...props} />);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
