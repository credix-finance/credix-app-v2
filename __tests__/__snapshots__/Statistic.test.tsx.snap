// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currency statistic 1`] = `
<div
  className="bg-transparent border border-solid border-darker rounded-[1px] font-sans h-36 w-min min-w-[16rem] md:w-full md:min-w-[12rem] ml-[21.5px] pr-5 flex items-center"
>
  <div
    className="bg-transparent ml-[-21.5px] relative"
  >
    <div
      className="left-5 absolute w-3 h-20 z-10 bg-credix-primary"
    />
    <div
      className="capitalize relative z-20"
    >
      TVL
    </div>
    <div
      className="text-6xl font-bold relative z-20"
    >
      14.8M
      <span
        className="text-sm font-normal"
      >
        USDC
      </span>
    </div>
  </div>
</div>
`;

exports[`Percentage statistic 1`] = `
<div
  className="bg-transparent border border-solid border-darker rounded-[1px] font-sans h-36 w-min min-w-[16rem] md:w-full md:min-w-[12rem] ml-[21.5px] pr-5 flex items-center"
>
  <div
    className="bg-transparent ml-[-21.5px] relative"
  >
    <div
      className="left-5 absolute w-3 h-20 z-10 bg-credix-primary"
    />
    <div
      className="capitalize relative z-20"
    >
      Estimated APY
    </div>
    <div
      className="text-6xl font-bold relative z-20"
    >
      13.8%
      <span
        className="text-sm font-normal"
      />
    </div>
  </div>
</div>
`;
